src/rest.cpp

```diff
@@ -181,13 +181,14 @@ static bool rest_headers(const std::any& context,
     headers.reserve(count);
     {
         LOCK(cs_main);
-        tip = ::ChainActive().Tip();
-        const CBlockIndex* pindex = g_chainman.m_blockman.LookupBlockIndex(hash);
-        while (pindex != nullptr && ::ChainActive().Contains(pindex)) {
+        ChainstateManager& chainman = EnsureChainman(context);
+        tip = chainman.ActiveChain().Tip();
+        const CBlockIndex* pindex = chainman.m_blockman.LookupBlockIndex(hash);
+        while (pindex != nullptr && chainman.ActiveChain().Contains(pindex)) {
             headers.push_back(pindex);
             if (headers.size() == (unsigned long)count)
                 break;
-            pindex = ::ChainActive().Next(pindex);
+            pindex = chainman.ActiveChain().Next(pindex);
```

Comment:
Maybe just grab `chainman.ActiveChain()` once above and then use it for each iteration.